src/Campaigns/resources/admin/components/CampaignDetailsPage/index.tsx

```diff
@@ -1,214 +1,139 @@
-import CampaignsApi from '../api';
+import {Campaign, GiveCampaignDetails} from './types';
 import {FormProvider, SubmitHandler, useForm} from 'react-hook-form';
-import {Tab, TabList, TabPanel, Tabs} from 'react-aria-components';
+import {Spinner} from '@givewp/components';
+import Tabs from './Tabs';
+
+import styles from './style.module.scss';
 
 declare const window: {
     GiveCampaignDetails: GiveCampaignDetails;
 } & Window;
 
-export function getGiveCampaignDetailsWindowData() {
-    return window.GiveCampaignDetails;
-}
-
-const {adminUrl, campaign, apiRoot, apiNonce} = getGiveCampaignDetailsWindowData();
-const API = new CampaignsApi({apiNonce, apiRoot});
-const tabs: CampaignDetailsTab[] = campaignDetailsTabs;
-
-export default function CampaignsDetailsPage() {
-    /**
-     * TABS LOGIC
-     */
-    const [activeTab, setActiveTab] = useState<CampaignDetailsTab>(tabs[0]);
-
-    const getTabFromURL = () => {
-        const urlParams = new URLSearchParams(window.location.search);
-        const tabId = urlParams.get('tab') || activeTab.id;
-        const tab = tabs.find((tab) => tab.id === tabId);
-        console.log('tab: ', tab);
-
-        return tab;
-    };
-
-    const handleTabNavigation = (tabId: string) => {
-        const newTab = tabs.find((tab) => tab.id === tabId);
+export default function CampaignsDetailsPage({campaignId}) {
 
-        if (!newTab) {
-            return;
-        }
-
-        // @ts-ignore
-        const url = new URL(window.location);
-        const urlParams = new URLSearchParams(url.search);
-
-        urlParams.set('tab', newTab.id);
-
-        const newUrl = `${url.pathname}?${urlParams.toString()}`;
-        window.history.pushState(null, activeTab.title, newUrl);
-
-        setActiveTab(newTab);
-    };
-
-    const handleUrlTabParamOnFirstLoad = () => {
-        // @ts-ignore
-        const url = new URL(window.location);
-        const urlParams = new URLSearchParams(url.search);
-
-        // Add the 'tab' parameter only if it's not in the URL yet
-        if (!urlParams.has('tab')) {
-            urlParams.set('tab', activeTab.id);
-            const newUrl = `${url.pathname}?${urlParams.toString()}`;
-            window.history.replaceState(null, activeTab.title, newUrl);
-        } else {
-            setActiveTab(getTabFromURL());
-        }
-    };
+    const [resolver, setResolver] = useState({});
 
     useEffect(() => {
-        handleUrlTabParamOnFirstLoad();
-
-        const handlePopState = () => {
-            console.log('handlePopState');
-            setActiveTab(getTabFromURL());
-        };
-
-        // Updates state based on URL when user navigates with "Back" or "Forward" buttons
-        window.addEventListener('popstate', handlePopState);
-
-        // Cleanup listener on unmount
-        return () => {
-            window.removeEventListener('popstate', handlePopState);
-        };
+        fetch(`/wp-json/give-api/v2/campaigns/${campaignId}`, {method: 'OPTIONS'})
```

Comment:
You'll want to use `@wordpress/api-fetch`. It's preconfigured to talk to the REST API. Then you'll drop the `/wp-json/` prefix (it's configurable via a filter, and may not work at all since pretty permalinks aren't guaranteed), since WP will speak to the right URL by default.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7547/commits/31a91bef3143fce84170bbb63a84e8184b635869